### PR TITLE
[LITE]Elementwise ops refactor

### PR DESCRIPTION
**Pull request by siju-samuel**

### CODE REVIEW

**Comment by siju-samuel** [comment]
@jdduke @achowdhery could you please help to review this. TIA

File: tensorflow/lite/kernels/elementwise.cc [hunk 1]
```diff
@@ -107,6 +116,59 @@ TfLiteStatus LogicalNotEval(TfLiteContext* context, TfLiteNode* node) {
...
+template <KernelType type>
+TfLiteStatus FloorEval(TfLiteContext* context, TfLiteNode* node) {
+  if (type == kReference) {
+    return EvalNumeric(context, node, [](float f) { return std::floor(f); });
+  } else if (type == kGenericOptimized) {
+    return EvalOptimized(context, node, optimized_ops::Floor);
+  } else {
+    return kTfLiteError;
+  }
+  return kTfLiteOk;
```
Comment (jdduke) [inline tensorflow/lite/kernels/elementwise.cc]: No need for return kTfLiteOk

File: tensorflow/lite/kernels/elementwise.cc [hunk 2]
```diff
@@ -107,6 +116,59 @@ TfLiteStatus LogicalNotEval(TfLiteContext* context, TfLiteNode* node) {
...
+template <KernelType type>
+TfLiteStatus CeilEval(TfLiteContext* context, TfLiteNode* node) {
+  if (type == kReference) {
+    return EvalNumeric(context, node, [](float f) { return std::ceil(f); });
+  } else if (type == kGenericOptimized) {
+    return EvalOptimized(context, node, optimized_ops::Ceil);
+  } else {
+    return kTfLiteError;
+  }
+  return kTfLiteOk;
```
Comment (jdduke) [inline tensorflow/lite/kernels/elementwise.cc]: No need for `reutnr kTfLiteOk`.

File: tensorflow/lite/kernels/elementwise.cc (hunk 1 above)
Comment (jdduke) [reply tensorflow/lite/kernels/elementwise.cc]: Alternatively, use a switch, with no default case, which will ensure you always handle the path. Same below.

**Review by jdduke — CHANGES_REQUESTED** [review]

**Comment by siju-samuel** [comment]
@jdduke Thanks a lot for the review, i have updated the code based on your comments. Could you please have a look again. TIA

File: tensorflow/lite/kernels/elementwise.cc [hunk 3]
```diff
@@ -107,6 +116,64 @@ TfLiteStatus LogicalNotEval(TfLiteContext* context, TfLiteNode* node) {
...
+  if ((diff < 0.5f) ||
+      ((diff == 0.5f) && (static_cast<int>(floor_val) % 2 == 0))) {
+    return floor_val;
+  } else {
+    return floor_val = floor_val + 1.0f;
+  }
+}
+
+TfLiteStatus RoundEval(TfLiteContext* context, TfLiteNode* node) {
+  return EvalNumeric(context, node, [](float f) { return RoundToNearest(f); });
```
Comment (jdduke) [inline tensorflow/lite/kernels/elementwise.cc]: Why the lambda? Just pass RoundtoNearest directly to EvalNumeric.

**Review by jdduke — DISMISSED** [review]

**Comment by rthadur** [comment]
@siju-samuel can you please resolve conflicts.

**Comment by girving** [comment]
I won't be able to do this review.

**Comment by siju-samuel** [comment]
@jdduke Gentle ping to review and approve once again. I rebased it due to the recent changes. Thanks in advance.

**Review by jdduke — APPROVED** [review]

File: tensorflow/lite/kernels/internal/reference/reference_ops.h [hunk 4]
```diff
@@ -2635,49 +2635,6 @@ T FloorMod(T input1, T input2) {
              : trunc_mod;
 }
 
-inline void Floor(const RuntimeShape& input_shape, const float* input_data,
```
Comment (jdduke) [inline tensorflow/lite/kernels/internal/reference/reference_ops.h]: For internal reasons, we need to keep the Floor/Ceil/Round implementations in this file. Can you please update accordingly? Thanks.

**Review by jdduke — CHANGES_REQUESTED** [review]

**Review by jdduke — DISMISSED** [review]

**Comment by rthadur** [comment]
@siju-samuel thank you , can you please check this build failures : 
https://source.cloud.google.com/results/invocations/da218a31-ea37-4298-b520-309de102e5f0/log
https://source.cloud.google.com/results/invocations/9a225bff-fc03-4596-b2a4-3f0ee7a68881/log

**Comment by siju-samuel** [comment]
@rthadur Sorry for the delay,  I could find that target `//tensorflow/python/keras/distribute:keras_dnn_correctness_test`
 is failing with and without my changes. My modifications are in lite folder only. 

Another build error showing is, which may not b the actual reason, which is also not in scope of my change
`
ERROR: /tmpfs/src/github/tensorflow/tensorflow/tools/api/golden/BUILD:3:1: invalid label '//tensorflow:tensorflow_py:__subpackages__' in element 0 of 'package' argument: invalid target name 'tensorflow_py:__subpackages__': target names may not contain ':'
ERROR: package contains errors: tensorflow/tools/api/golden
Loading: 412 packages loaded
    currently loading: tensorflow/cc/saved_model/python ... (54 packages)
Loading: 462 packages loaded
    currently loading: tensorflow/core/kernels ... (4 packages)
ERROR: error loading package 'tensorflow/tools/api/golden': Package 'tensorflow/tools/api/golden' contains errors
`

Could you please help me find the reason. Thanks in advance.

**Comment by rthadur** [comment]
I will pull this in and try to merge

**Comment by siju-samuel** [comment]
Rebased due to conflict.

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by gbaned** [comment]
@siju-samuel  Could you please resolve the conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
